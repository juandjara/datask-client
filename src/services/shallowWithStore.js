// https://medium.com/@visualskyrim/test-your-redux-container-with-enzyme-a0e10c0574ec
import { shallow } from 'enzyme';

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

export default shallowWithStore;