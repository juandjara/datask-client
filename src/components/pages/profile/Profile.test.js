import React from 'react'
import ProfileConnected, {Profile} from './Profile'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import shallowWithStore from 'services/shallowWithStore'
import {createMockStore} from 'redux-test-utils'

const profile = {
  name: 'Juan D.',
  surname: 'Jara',
  email: 'juanorigami@gmail.com'
}

describe('Profile', () => {
  it('renders with store without crashing', () => {
    const state = {profile: {}}
    const store = createMockStore(state)
    shallowWithStore(<ProfileConnected />, store)
  })
  it('matches snapshot with profile', () => {
    const component = renderer.create(
      <Profile 
        profile={profile}
        saveProfile={jest.fn()}
        fetchProfile={jest.fn()}
        updateProfileField={jest.fn()}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('matches snapshot with loading', () => {
    const component = renderer.create(
      <Profile 
        profile={{loading: true}}
        saveProfile={jest.fn()}
        fetchProfile={jest.fn()}
        updateProfileField={jest.fn()}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('calls fetchProfile on mount', () => {
    const component = renderer.create(
      <Profile 
        profile={profile}
        saveProfile={jest.fn()}
        fetchProfile={jest.fn()}
        updateProfileField={jest.fn()}
      />
    )
    const {props} = component.getInstance()
    expect(props.fetchProfile).toHaveBeenCalled()
  })
  it('calls saveProfile on submit', () => {
    const component = shallow(
      <Profile 
        profile={profile}
        saveProfile={jest.fn()}
        fetchProfile={jest.fn()}
        updateProfileField={jest.fn()}
      />
    )
    const props = component.instance().props
    component.find('form').props().onSubmit({
      preventDefault: () => {}
    })
    expect(props.saveProfile).toHaveBeenCalled()
  })
  it('calls updateProfileField on change', () => {
    const component = mount(
      <Profile 
        profile={profile}
        saveProfile={jest.fn()}
        fetchProfile={jest.fn()}
        updateProfileField={jest.fn()}
      />
    )
    const props = component.instance().props
    component.find('input').first().props().onChange({
      target: {name: 'someName', value: 'some text'}
    })
    expect(props.updateProfileField).toHaveBeenCalled()
  })
})
