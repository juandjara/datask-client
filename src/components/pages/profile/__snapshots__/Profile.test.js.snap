// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches snapshot with loading 1`] = `
<div
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <h2
    className="type-headline"
  >
    Mi cuenta
  </h2>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexFlow": "row",
        "justifyContent": undefined,
      }
    }
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className=""
      data-react-toolbox="progress-bar"
      disabled={undefined}
    >
      <svg
        className={undefined}
        viewBox="0 0 60 60"
      >
        <circle
          className={undefined}
          cx="30"
          cy="30"
          r="25"
          style={undefined}
        />
      </svg>
    </div>
    <span
      style={
        Object {
          "marginLeft": "1em",
        }
      }
    >
      Cargando ...
    </span>
  </div>
  <div
    className="color-error"
  >
    <pre />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="undefined"
      data-react-toolbox="input"
    >
      <input
        className=""
        defaultValue={undefined}
        disabled={false}
        maxLength={undefined}
        name="email"
        onChange={[Function]}
        onKeyPress={undefined}
        required={false}
        role="input"
        type="email"
        value=""
      />
      <span
        aria-label=""
        className="material-icons"
        data-react-toolbox="font-icon"
      >
        email
      </span>
      <span
        className={undefined}
      />
      <label
        className=""
      >
        Email
      </label>
    </div>
    <div
      className="sc-bdVaJa eRcGeI"
    >
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="name"
          onChange={[Function]}
          onKeyPress={undefined}
          required={false}
          role="input"
          type="text"
          value=""
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          person_outline
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Nombre
        </label>
      </div>
      <div
        className=""
        data-react-toolbox="input"
      >
        <input
          className=""
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="surname"
          onChange={[Function]}
          onKeyPress={undefined}
          required={false}
          role="input"
          type="text"
          value=""
        />
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Apellidos
        </label>
      </div>
    </div>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Guardar
    </button>
  </form>
</div>
`;

exports[`Profile matches snapshot with profile 1`] = `
<div
  style={
    Object {
      "padding": "1em",
    }
  }
>
  <h2
    className="type-headline"
  >
    Mi cuenta
  </h2>
  <div
    className="color-error"
  >
    <pre />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="undefined"
      data-react-toolbox="input"
    >
      <input
        className="undefined"
        defaultValue={undefined}
        disabled={false}
        maxLength={undefined}
        name="email"
        onChange={[Function]}
        onKeyPress={undefined}
        required={false}
        role="input"
        type="email"
        value="juanorigami@gmail.com"
      />
      <span
        aria-label=""
        className="material-icons"
        data-react-toolbox="font-icon"
      >
        email
      </span>
      <span
        className={undefined}
      />
      <label
        className=""
      >
        Email
      </label>
    </div>
    <div
      className="sc-bdVaJa eRcGeI"
    >
      <div
        className="undefined"
        data-react-toolbox="input"
      >
        <input
          className="undefined"
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="name"
          onChange={[Function]}
          onKeyPress={undefined}
          required={false}
          role="input"
          type="text"
          value="Juan D."
        />
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          person_outline
        </span>
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Nombre
        </label>
      </div>
      <div
        className=""
        data-react-toolbox="input"
      >
        <input
          className="undefined"
          defaultValue={undefined}
          disabled={false}
          maxLength={undefined}
          name="surname"
          onChange={[Function]}
          onKeyPress={undefined}
          required={false}
          role="input"
          type="text"
          value="Jara"
        />
        <span
          className={undefined}
        />
        <label
          className=""
        >
          Apellidos
        </label>
      </div>
    </div>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Guardar
    </button>
  </form>
</div>
`;
